File: showcase/password-retry.c
#include <stdio.h>
#include <string.h>

int main()
{
  char password[20] = "password123"; // Set the correct password
  char input[20];                    // Variable to store user input

  int attempts = 3; // Number of attempts allowed

  while (attempts > 0)
  {
    printf("Enter the password: ");
    scanf("%s", input);

    // Check if the input matches the correct password
    if (strcmp(input, password) == 0)
    {
      printf("Access Granted\n");
      break;
    }
    else
    {
      attempts--;
      printf("Access Denied\n");
      printf("Attempts remaining: %d\n", attempts);
    }
  }

  // Check if all attempts are used
  if (attempts == 0)
  {
    printf("Access Denied\n");
  }

  return 0;
}

File: showcase/password-retry.php
<?php

$correctPassword = "password"; // Set the correct password here
$maxAttempts = 3; // Set the maximum number of attempts here

$attempts = 0; // Initialize the attempts counter

while ($attempts < $maxAttempts) {
    $attempts++;

    $enteredPassword = readline("Enter the password: "); // Prompt the user to enter the password

    if ($enteredPassword === $correctPassword) {
        echo "Access Granted" . PHP_EOL; // Display "Access Granted" message
        break; // Exit the loop if the password is correct
    } else {
        echo "Access Denied" . PHP_EOL; // Display "Access Denied" message
    }
}

if ($attempts === $maxAttempts) {
    echo "Max attempts reached. Access Denied" . PHP_EOL; // Display "Access Denied" message if maximum attempts reached
}

echo "\n";